onyo/commands/edit.py

```diff
@@ -3,39 +3,79 @@
 import sys
 
 from onyo.lib import Repo, InvalidOnyoRepoError
+from pathlib import Path
+from ruamel.yaml import YAML  # pyre-ignore[21]
+
 from onyo.utils import (
-    run_cmd,
-    edit_file
+    get_editor
 )
 
 logging.basicConfig()
 log = logging.getLogger('onyo')
 
 
-def prepare_arguments(sources, onyo_root):
-    problem_str = ""
+def edit_asset(editor: str, asset: Path) -> bool:
+    """
+    Open an existing asset with an editor. After changes are made, check the
+    asset for validity and check its YAML syntax. If valid, write the changes,
+    otherwise open a dialog and ask the user if the asset should be corrected
+    or the changes discarded.
+
+    Returns True on success.
+    """
+    further_editing = 'y'
+    while further_editing in ['y', 'Y', 'yes']:
+        os.system(f'{editor} "{asset}"')
+
+        yaml_error = ""
+        try:
+            YAML().load(asset)
+        except Exception:
+            yaml_error = f"{asset} has invalid YAML syntax."
+
+        # TODO: add asset validity here
+        validity_error = ""
+
+        if not any([yaml_error, validity_error]):
+            return True
+
+        else:
+            if yaml_error:
+                print(yaml_error)
+            if validity_error:
+                print(validity_error)
+            while True:
+                further_editing = input("Continue editing? No discards changes. (y/N) ")
+                if further_editing in ['y', 'Y', 'yes', 'n', 'N', 'no']:
+                    break
+
+        if further_editing in ['n', 'N', 'no']:
+            return False
+    return False
+
+
+def prepare_arguments(sources: list, repo: Repo) -> list:
```

Comment:
What does this function do?

I assumed it prepared the arguments, but it seems to return a list of assets.

A better name and a docstring would help reduce the ambiguity.

Please also specify the types inside the lists when possible. Such as:
```
list[Path]
```